torch/csrc/jit/codegen/cuda/interface.cpp

```diff
@@ -24,6 +24,103 @@ namespace cuda {
 
 static std::atomic<bool> cuda_fusion_guard_mode{true};
 
+// There are 3 sources of information on whether to enable nvfuser:
+// 1. assigned value from setEnabled() - takes precendence if it has been set
+// 2. value from environment variable - only used if setEnabled() is unset
+// 3. default value - used if both 1 and 2 are unset.
+//
+// If 1 or 2 tries to enable nvfuser when it cannot be enabled (e.g. cuda not
+// available), then an error will be thrown. The default will not error.
+class NVFuserEnabler {
+ private:
+  c10::optional<bool> runtime_assigned_fuser_enabled_ = c10::nullopt;
+  std::once_flag enabled_check_flag_;
+  std::mutex mutex_;
+
+  static bool nvfuserCanBeEnabled() {
```

Comment:
nitpick: we are just returning `false` for rocm here. Might as well hard code that in the macro.